### ISSUE TITLE
[BUG][Functional] Fail to create peering connection

### ISSUE THREAD

**Issue by rverma-nsl** [aws-samples/aws-secure-environment-accelerator#740]
Bug reports which fail to provide the required information will be closed without action.

**Required Basic Info**
- Accelerator Version:  (eg. v1.3.3)

**Describe the bug**
While creating 2 peering connections within account the ASEA throws below error. It seems to happen when I have a peering connection within one region and then tried again for another region within same account act as both requestor & acceptor.

```
Error: There is already a Construct with name 'PeeringRole' in AccountStack [SharedNetworkPhase1]
47 | at Node.addChild (/app/node_modules/.pnpm/constructs@3.3.71/node_modules/constructs/src/construct.ts:534:13)
48 | at new Node (/app/node_modules/.pnpm/constructs@3.3.71/node_modules/constructs/src/construct.ts:75:22)
49 | at new ConstructNode (/app/node_modules/.pnpm/@aws-cdk/core@1.85.0/node_modules/@aws-cdk/core/lib/construct-compat.ts:277:24)
50 | at Object.createNode (/app/node_modules/.pnpm/@aws-cdk/core@1.85.0/node_modules/@aws-cdk/core/lib/construct-compat.ts:71:11)
51 | at new Construct (/app/node_modules/.pnpm/constructs@3.3.71/node_modules/constructs/src/construct.ts:575:26)
52 | at new Construct (/app/node_modules/.pnpm/@aws-cdk/core@1.85.0/node_modules/@aws-cdk/core/lib/construct-compat.ts:68:5)
53 | at new Resource (/app/node_modules/.pnpm/@aws-cdk/core@1.85.0/node_modules/@aws-cdk/core/lib/resource.ts:115:5)
54 | at new Role (/app/node_modules/.pnpm/@aws-cdk/aws-iam@1.85.0/node_modules/@aws-cdk/aws-iam/lib/role.ts:306:5)
55 | at createIamRoleForPCXAcceptence (/app/src/deployments/cdk/src/apps/phase-1.ts:128:25)
56 | at deploy (/app/src/deployments/cdk/src/apps/phase-1.ts:251:9)
57
```

**Required files**

[config.txt](https://github.com/aws-samples/aws-secure-environment-accelerator/files/6528504/config.txt)

**Comment by rverma-nsl** [comment]
@Brian969, the culprit is https://github.com/aws-samples/aws-secure-environment-accelerator/blob/c3ce485a2c01f061d5f2073c84ea236495e5ef0d/src/deployments/cdk/src/apps/phase-1.ts#L251. Is there any quick fix you recommend here. Seems like this is related to when running in parallel. I tried appending with region/account/etc to stack id but no luck

**Comment by rverma-nsl** [comment]
Also tried setting up vpc peering connection to endpoint in place of tgw to multiple vpc. The peering routing tables are all juggled up. E.g. In vpc1 it is trying to put peering connection referenced to vpc2 etc...

**Comment by Brian969** [comment]
Hi,
- on line number 2127 and line 2521 you have the same VPC defined twice.  We do not support more than 1 VPC with the same name and will cause issues - can you start by fixing this and getting back to us? (it could be the cause of your issue, they both have pcx connections defined).

**Comment by Brian969** [comment]
On the routing issue, I see many route tables with the same name, in the same account.  Not sure if this is a problem or not, but, good chance we cannot tell them apart.  Can you also change to unique route table names? (i.e. these names are duplicated: EndpointVPC_Common, Common, Internal, default).

**Comment by rverma-nsl** [comment]
Hi @Brian969 the issue with peeringrole persisted even when the duplicated vpc was removed. The routing table i have given uniquenames. I am not sure whether its routetable cause I am suspecting you are anticipating some order of route destination. If you look at below section, this is failing cause in endpoint vpc i have routetable entries as
```
route-tables:
  - name: EndpointVPC_Common
    routes:
      - destination:
          account: shared-network
          vpc: SharedServices
          subnet: Internal
        target: pcx
      - destination:
          account: shared-network
          vpc: Non-Production
          subnet: Transit
        target: pcx
      - destination:
          account: shared-network
          vpc: Production
          subnet: Transit
        target: pcx
```
while in vpc i have ordering of 
1. Nonprod
2. SharedServices

**Comment by rverma-nsl** [comment]
Also I am hoping we should be able to define pcx routes as 
```
- destination:
          account: shared-network
          vpc: SharedServices-useast1
          subnet: Public <-- remove this line to create route for complete vpc cidr
        target: pcx
```

**Comment by rverma-nsl** [comment]
Providing updated config.json 
[config.txt](https://github.com/aws-samples/aws-secure-environment-accelerator/files/6543441/config.txt)
Few other issues observed:
1. The architecture allows only single routetable entry to either pcx or TGW. Have to add commits https://github.com/nslhb/aws-secure-environment-accelerator/commit/0b07461b4ad5d5079ed3e1ef02bc5d83e51ce535
& https://github.com/nslhb/aws-secure-environment-accelerator/commit/7185a0967da2d12c0f78e4b8ae2b0570951291a7 to fix the same.

Still error with already a Construct with name 'PeeringRole' exists. Couldn't figure out the resolution of the same.

**Comment by Brian969** [comment]
Thanks for the routing fixes - I'll get someone to take a look at the role issue.

**Comment by Brian969** [comment]
All these fixes merged with main (currently testing).  We added some extra code to not break existing clients/keep the construct name the same for existing deployments on the TGW route issue.

**Comment by Brian969** [comment]
merged w/main, will ship in v1.3.4
